### PR TITLE
Add endpoints to support Discord printing command

### PR DESCRIPTION
**Pull request by phuvu0107**
The new discord bot feature will allow users who connected their discord account to see web account to print files via dos. But in order to do that, we need to communicate with Core-v4 and these APIs will allows us to do exactly that.

New endpoints:
- /Users/checkDiscordID (using an API key, checks if user is connected to web account)
- /Printer/pushDiscordPDFToSqs (using an API key, pushes incoming file url from discord to SQS)

### CODE REVIEW

File: api/peripheral_api/routes/Printer.js
Comment (neelp03) [inline api/peripheral_api/routes/Printer.js:10]:
```suggestion
const { s3BucketKeys, printingS3Bucket }  = require('../../config/config.json');
```

You can import multiple json properties all together like this

File: api/peripheral_api/routes/Printer.js
Comment (neelp03) [inline api/peripheral_api/routes/Printer.js:51]:
```suggestion
```

File: api/main_endpoints/server.js
Comment (neelp03) [inline api/main_endpoints/server.js:8]:
```suggestion
  const mainEndpointServer = new SceHttpServer(API_ENDPOINTS, 8080, '/mainendpoints/');
```

File: api/peripheral_api/routes/Printer.js
Comment (neelp03) [inline api/peripheral_api/routes/Printer.js:96]: rename it to something like validateDiscordReq or like incomingDiscordReq, just something that describes what the function does

**Review by neelp03 — CHANGES_REQUESTED** [review]

File: api/main_endpoints/routes/User.js
Comment (evanugarte) [inline api/main_endpoints/routes/User.js:275]: you can revert all the other changes, you only need this

File: api/main_endpoints/server.js
Comment (neelp03) [inline api/main_endpoints/server.js:9]: instead of disabling eslint, just put the other half on a new line and indent

File: api/main_endpoints/routes/User.js
Comment (neelp03) [inline api/main_endpoints/routes/User.js:11]: are we even using registerUser anywhere in here??

File: api/peripheral_api/routes/Printer.js
Comment (neelp03) [inline api/peripheral_api/routes/Printer.js:109]: can put it all on one line
```suggestion
    MessageBody: JSON.stringify({ fileURL }),
```

**Review by neelp03 — CHANGES_REQUESTED** [review]

File: api/peripheral_api/routes/Printer.js
Comment (neelp03) [inline api/peripheral_api/routes/Printer.js:16]: You can just import these with the others on top on line 9

File: api/main_endpoints/routes/User.js
Comment (neelp03) [inline api/main_endpoints/routes/User.js:11]: put these on same line

**Review by neelp03 — CHANGES_REQUESTED** [review]

**Review by neelp03 — APPROVED** [review]
Nice! One step closer to printing via discord

File: api/peripheral_api/routes/Printer.js
Comment (evanugarte) [inline api/peripheral_api/routes/Printer.js:94]: maybe
```suggestion
router.post('/pushDiscordPdfToSqs', async (req, res) => {
```

?

File: api/main_endpoints/routes/User.js
Comment (evanugarte) [inline api/main_endpoints/routes/User.js:281]:
```suggestion
  User.findOne({ discordID }, (error, result) => {
```

File: api/main_endpoints/routes/User.js
Comment (evanugarte) [inline api/main_endpoints/routes/User.js:288]:
```suggestion
    let result = OK;
    if (error) {
      result = BAD_REQUEST;
    } else if (!result) {
      result = NOT_FOUND;
    }
    return res.sendStatus(result);
```

File: api/peripheral_api/routes/Printer.js
Comment (evanugarte) [inline api/peripheral_api/routes/Printer.js:111]: we should use [`SceSqsApiHandler`](https://github.com/SCE-Development/Core-v4/blob/2c5fd8b7b546d90a09d9be8d29b7e3b8054b0afa/api/peripheral_api/util/SceSqsApiHandler.js#L10)

open an issue for it if were not doing it rn

**Review by evanugarte — APPROVED** [review]
need tests for this, we dont have it for `sendPrintRequest` either. an issue for this would b good

**Review by neelp03 — CHANGES_REQUESTED** [review]
Need api tests for the new endpoint in peripheral api

**Review by neelp03 — APPROVED** [review]
Thanks for adding the documentation. Everything else looks good to me. We might need to change some stuff if we decide to implement the pages counted on discord side differently.

**Comment by neelp03** [comment]
DONT MERGE YET

File: api/main_endpoints/server.js
Comment (evanugarte) [inline api/main_endpoints/server.js:8]:
```suggestion
  const mainEndpointServer =  new SceHttpServer(API_ENDPOINTS, 8080);
```

File: api/config/config.example.json
Comment (evanugarte) [inline api/config/config.example.json:49]: if we build out more features between our bot and core-v4 we will use this key again, maybe dont call it printing and instead

```suggestion
  "DISCORD_COREV4_KEY":"xxxxxxxxxxxxxxxx"
```


u can also skip and leave for when we actually do this stuff

File: api/peripheral_api/routes/Printer.js
Comment (evanugarte) [inline api/peripheral_api/routes/Printer.js:24]: i would rather have this be initialized in the API handler instead of being global, up to you, also use camelcase (`sqsHandler`)

File: api/peripheral_api/routes/Printer.js
Comment (evanugarte) [inline api/peripheral_api/routes/Printer.js:87]:
```suggestion
    copies,
```

File: api/peripheral_api/routes/Printer.js
Comment (evanugarte) [inline api/peripheral_api/routes/Printer.js:90]: lint this file one time before merging

```suggestion
  const result = SqsHandler.pushMessageToQueue(data);
```

File: api/peripheral_api/routes/Printer.js
Comment (evanugarte) [inline api/peripheral_api/routes/Printer.js:104]: can we make this more descriptive? like
```suggestion
  const result = await SqsHandler.pushMessageToQueue({discordFileUrl});
```

that way on the quasar side we can do
```js
if (sqsMessage.discordFileUrl) {
  // handle special discord logic
} else {
  // default quasar logic
}
```

File: api/util/token-verification.js
Comment (evanugarte) [inline api/util/token-verification.js:44]:
```suggestion
function checkDiscordKey(apiKey) {
  return apiKey === DISCORD_PRINTING_KEY;
}
```

File: test/api/RFID.js
Comment (evanugarte) [inline test/api/RFID.js:110]: why

File: test/api/User.js
Comment (evanugarte) [inline test/api/User.js:289]: ?

File: test/api/User.js
Comment (evanugarte) [inline test/api/User.js:231]: why is this deleted

File: test/util/mocks/TokenValidFunctions.js
Comment (evanugarte) [inline test/util/mocks/TokenValidFunctions.js:16]: token and discord are different things, since only 1 api rn cares about discord api keys, move this back into the user.js test file

File: test/api/User.js
Comment (neelp03) [reply test/api/User.js:231]: Its not, we just moved it to the bottom

File: api/config/config.example.json
Comment (neelp03) [inline api/config/config.example.json:45]: Why these??

File: api/main_endpoints/routes/User.js
Comment (neelp03) [inline api/main_endpoints/routes/User.js:261]: delete these random comments, and instead if you want to, you can just explain how it works in multi-line comments

File: api/main_endpoints/routes/User.js
Comment (neelp03) [inline api/main_endpoints/routes/User.js:284]: instead of disabling eslint for that line, you can just split it up into multiple lines like this:

```js
  }
  // eslint-disable-next-line max-len
  User.updateOne( { discordID }, {pagesPrinted: newPagesPrinted},
    (error, result) => {
```
something like that

File: api/main_endpoints/server.js
Comment (neelp03) [inline api/main_endpoints/server.js:7]: some random change

File: test/util/mocks/DiscordApiFunction.js
Comment (neelp03) [inline test/util/mocks/DiscordApiFunction.js:1]: Lets call this DiscordValidation or something

File: test/util/mocks/DiscordApiFunction.js
Comment (neelp03) [inline test/util/mocks/DiscordApiFunction.js:30]: descriptions need to be updated

**Review by neelp03 — CHANGES_REQUESTED** [review]

**Review by ShaivPandya — APPROVED** [review]

**Review by neelp03 — APPROVED** [review]

**Review by neelp03 — APPROVED** [review]
LGTM but idky there are changes for the health check api. Might wanna double check that.

File: api/util/token-verification.js
Comment (evanugarte) [inline api/util/token-verification.js:40]: optional
```suggestion
function discordKeyValid(apiKey) {
```

File: api/main_endpoints/util/token-functions.js
Comment (evanugarte) [inline api/main_endpoints/util/token-functions.js:2]: undo this change

File: api/main_endpoints/routes/User.js
Comment (evanugarte) [inline api/main_endpoints/routes/User.js:254]: since we capitalize ID in discordID, we could do the same capitalization here. up to you

File: test/api/User.js
Comment (evanugarte) [inline test/api/User.js:329]: maybe compare result with the user you created to make sure they match

**Pull request merged**
